mantidproject/mantid
# Refactor plotting in Muon and Frequency domain GUIs

**Description of work.**
This PR refactors and reorganises the plotting code within the Muon Analysis and Frequencey Domain GUIS. The primary goals of this PR was to:

- Reduce duplication in the plotting code (as raised in #28406)
- Create a manageable code-base for plotting in the GUI as previously all the code was condensed into three files.
- Reduce dependence on a specific implementation of plotting by coding to an interface. This is likely to be beneficial as it is envisaged that frequency domain analysis will required some of its own plotting logic. 
- Base the view in the tab off a UI file so it can be easily changed. 
- As a side effect of restructuring, fixes multiple fits being plotted #28502

Left to do

- [x]  Plotting of fits (both single and sequential)
- [x] Connect QuickEditWidget with plot toolbar




**To test:**
Open the Muon Analysis 2 GUI

1. Load some data e.g MUSR62260
2. Check the following functionality works (in no particular order):
a. Changing plot type
b. External plotting
c. Tiled plotting (by groups and Runs (load some more data if necessary)
d. Grouping tab (add and remove groups)
e. Rebin (options are changed in the home tab) 
f. Plot raw should switch between rebinned and raw data
g. The functionality in #28406 should be reproducible. 




Fixes #28502. 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
